robot/src/main/java/us/ilite/robot/modules/Limelight.java

```diff
@@ -0,0 +1,250 @@
+package us.ilite.robot.modules;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import us.ilite.common.config.SystemSettings;
+
+public class Limelight extends Module {
+
+    private final NetworkTable mTable = NetworkTableInstance.getDefault().getTable("limelight");
+    public TargetingData mCurrentTarget = null;
+
+    @Override
+    public void modeInit(double pNow) {
+
+    }
+
+    @Override
+    public void periodicInput(double pNow) {
+
+    }
+
+    @Override
+    public void update(double pNow) {
+        mCurrentTarget = new TargetingData();
+    }
+
+    @Override
+    public void shutdown(double pNow) {
+
+    }
+
+    public enum LedMode {
+        NO_CHANGE,
+        LED_OFF,
+        LED_BLINK,
+        LED_ON;
+    }
+
+    public enum Stream {
+        STANDARD,
+        PIP_MAIN,
+        PIP_SECONDARY;
+    }
+
+    public void setCamMode(boolean pMode) {
+        mTable.getEntry("camMode").setBoolean(pMode);
+    }
+
+    public void setLedMode(LedMode pMode) {
+        mTable.getEntry("ledMode").setNumber(pMode.ordinal());
+    }
+
+    public void setPipeline(int pipeline) {
+        mTable.getEntry("pipeline").setNumber(pipeline);
+    }
+
+    public void setSnapshot(boolean snapshot) {
+        mTable.getEntry("snapshot").setBoolean(snapshot);
+    }
+
+    public void setStream(Stream stream) { 
+        mTable.getEntry("stream").setNumber(stream.ordinal());
+    }
+
+    
+
+    public class TargetingData {
+        public final boolean tv;
+        public final double tx, ty, ta, ts, tl, tshort, tlong, thoriz, tvert;
+
```

Comment:
For the Targetting Data itself, it may be beneficial to make this more generic than just the limelight as part of a 2nd module.  For example, we would make an interface that returned tx, tv, ty, ta, (etc), and the Limelight class would implement that interface.  In the implementation, the limelight class would pull from NT.  Then the 2nd module would use the LL data (via the generic interface) to do the calculations.

This would allow us to use additional cameras with opencv, multiple LL's, a Pixy, a Zed (one day...), etc.